### PR TITLE
chore: Keep Nuxt's 'asyncData' and 'fetch' with 'data'

### PR DESCRIPTION
**Pull request by rchl**
Change the order of asyncData and fetch properties to be next
to data property. All those are function-wise more or less
equivalent so IMO it makes sense to keep them together. asyncData
and fetch are primarily for setting up component's data.

@manniL

### CODE REVIEW

**Comment by manniL** [comment]
I'd love to get feedback about this from @pi0 and @clark ☺️

File: lib/rules/order-in-components.js [hunk 1]
```diff
@@ -19,11 +19,9 @@ const defaultOrder = [
   'inheritAttrs',
   'model',
   ['props', 'propsData'],
-  'data',
+  ['asyncData', 'fetch', 'data'],
```
Comment (pi0) [inline lib/rules/order-in-components.js]: I suggest moving fetch to the top of `asyncData` because it is more context less

**Comment by rchl** [comment]
@pi0 but as I understand, having asyncData and fetch in same array means the order can be freely chosen between those. Then it shouldn't be need for moving them around?

File: lib/rules/order-in-components.js (hunk 1 above)
Comment (rchl) [reply lib/rules/order-in-components.js]: @pi0 but as I understand, having asyncData and fetch in same array means the order can be freely chosen between those. Then it shouldn't be need for moving them around?

Comment (rchl) [reply lib/rules/order-in-components.js]: OK, I have actually updated to have them in explicit order.

**Comment by rchl** [comment]
Ping

**Comment by manniL** [comment]
@rchl I forgot: could we had `head` somewhere too?

**Comment by rchl** [comment]
Added after props and before data related props. Scream if you disagree.

**Comment by manniL** [comment]
@rchl For me it's ok. Anyway, would it make sense at the end, as you can use methods & stuff in there too?

**Comment by rchl** [comment]
True. How about:
```
  'methods',
  'head',
  ['template', 'render'],
```
At the very end would work for me too but this possibly makes a bit more sense.

**Comment by manniL** [comment]
@rchl Agree with the proposed positioning ☺️

**Comment by rchl** [comment]
Seems ready to merge?

**Comment by rchl** [comment]
Merge please. :)

**Comment by manniL** [comment]
cc @michalsnik @mysticatea :relieved:

**Comment by vinayakkulkarni** [comment]
OK, this messed up my code. lol

```bash
/Users/vinayak/Development/Work/frontend/pages/signup.vue
  101:3  warning  The "methods" property should be above the "head" property on line 90  vue/order-in-components

✖ 145 problems (0 errors, 145 warnings)
```

**Comment by rchl** [comment]
@vinayakkulkarni it's unfortunate that this fix was lingering for so long. You have an option to either override the rule in your eslint configuration or fix automatically with eslint --fix

**Pull request merged**